### DISCUSSION TITLE
Emqx not terminate the connection after default 15 min CRL check after the cert is revoked.

### DISCUSSION THREAD

**Discussion by sc-sityad** [emqx/emqx#10956] [Q&A] [answered]
- I am using emqx 5.026 version in which there is only option to enabel the crl check which checks the CA authority certificate if it has the CRL list and checks if it revoked or not. 
- But I am using small step for certificate generation (CA) which does not support the CRLs and OCSP stapling. 
- I wanted to manage the CRLs with mongodb(with ssl) and integrate with emqx, How to do that can anyone guide me with detailed explanation?

**Comment by zmstone** [comment]
Hi @sc-sityad 
The CRL feature in EMQX is provided by Erlang/OTP.
It only supports HTTP endpoint to download the data at the moment.

If you want to swap in a download from mongodb instead, you'll need to dig into the Erlang/OTP code base.
I'm afraid I don't know this part of OTP good enough to provide more pointers.

**Comment by sc-sityad** [comment]
Hi @zmstone @lafirest 
- Ok I use http endpoint. In emqx 5.0.26 documentation there is only crl check enabled. 
- Can U explain how to give the http url. and In http url data is in which format or it should be embedded in client certificate. Since I am not using CRL or OCSP stapling in certificate because I use small step.
-  But I can add the revoked serial number in http url. But I don't know the format of data in the http url.
- Which plugins were reponsible for CRL feature in EMQX.

**Comment by sc-sityad** [comment]
Hi @zmstone 

- How to set the time for CRL checking period manually in emqx 5.0.26. 
- There is only one check for enable the CRL.

**Comment by zmstone** [comment]
The URL of the CRL is encoded in the client certificate.
You can run this command to inspect it `openssl x509 -in cert.crt -noout -text | grep crl`

**Comment by zmstone** [comment]
> Which plugins were reponsible for CRL feature in EMQX

This is a standard feature, there is no need to enable any plugin to enable CRL checks.

**Comment by sc-sityad** [comment]
Hi @zmstone 
How to set the time (configure the parameters ) for CRL checking period manually in emqx 5.0.26 ? I have added the documentation of emqx 4.4 version. I need to configure this  refresh time in the below image.
![Screenshot from 2023-06-08 15-28-59](https://github.com/emqx/emqx/assets/110105510/092e6966-b133-4f6a-b226-4c13e2b404d6)

**Reply by zmstone** [reply]
CRL cache urls is no longer needed to be configured in v5 (because it can be a dynamic list at runtime).

The other configs are not super critical, as the default values should work for most of the cases.

They are available in v5, but not documented at the current release:
https://github.com/emqx/emqx/blob/22df275596235c3415216e47a14038478b94894c/apps/emqx/src/emqx_schema.erl#L896-L919
This part of config subject to refactoring in future releases, but fat chance it will be released as it is
--- this is why we have not decided to make them public yet.

**Comment by sc-sityad** [comment]
Hi @zmstone 
I have set this in emqx.conf `EMQX_CRL_CACHE__REFRESH_INTERVAL=5m `
But it did not work, and the default 15 min is not workink also. After the certificate is revoked, the connection is still there.

**Reply by zmstone** [reply, marked as answer]
Hi @sc-sityad 

Please note that CRL check is only performed during TLS handshake.
Nothing happens to already established connections.

**Comment by zmstone** [comment]
Hi @sc-sityad 

Sorry for missing the point in the title earlier.
I think the issue described in the title has been addressed in my latest reply.
I'm going to mark that one as an answer and close this discussion.
For other subjects, it's better to be discussed in a different thread.

And thank you for using github discussions, if we are not responsive, you can ping us in in the discord channel,
which will work better than opening a issue with the same subject. :P

**Reply by zmstone** [reply]
invite like to discord: https://discord.gg/aKvyvUczrc
